src/bacnet/bacdef.h

```diff
@@ -187,6 +195,24 @@ typedef struct BACnet_Object_Id {
     uint32_t instance;
 } BACNET_OBJECT_ID;
 
+/* This Struct is for initialisation info coming from Elixir 
+*/
+typedef struct BACnet_Object_Init_s {
+    // Instance works as an index to the object within the type, the type number is not included in this value.
+    uint32_t Object_Instance;
+    //BACNET_CHARACTER_STRING Object_Name;
```

Comment:
Why? We could properly initialise the variable BACNET_CHARACTER_STRING type calling the appropriate API from bacnet NIF.

Reply:
it was just an extra layer of complication when all `BACNET_OBJECT_INIT_T` needed to do was match the passed in `{object_name, object_index, description, unit}`. It keeps that part of the init process simpler, without having to init the fields within `BACNET_CHARACTER_STRING`.
However, if we do need to support UTF8 then we'll need to handle that as well in the code, as currently we assume it's always simple ASCII. I haven't played around with charlist and binary conversion through the NIF recently.